### PR TITLE
Twitter get DM fixed

### PR DESCRIPTION
**Pull request by bakayim**
Twitter get dm functionality is fixed accordingly new API.

Possible usage is:

```
next_cursor = None
while next_cursor is not None:
	if next_cursor is not None:
	    direct_messages = connection.direct_messages(cursor=next_cursor, count=50)
	else:
	    direct_messages = connection.direct_messages(count=50)
	try:
	    next_cursor = direct_messages[0].next_cursor
	except Exception:
	    next_cursor = None

	# Do your stuff
```

### CODE REVIEW

File: tweepy/api.py [hunk 1]
```diff
@@ -412,9 +412,9 @@ def direct_messages(self):
         """
         return bind_api(
             api=self,
-            path='/direct_messages.json',
+            path='/direct_messages/events/list.json',
             payload_type='direct_message', payload_list=True,
-            allowed_param=['since_id', 'max_id', 'count', 'full_text'],
+            allowed_param=['since_id', 'cursor', 'count', 'full_text'],
```
Comment (AgoraSecurity) [inline tweepy/api.py]: I understand that there are only 2 parameters allowed:
count & cursor 

Source: https://developer.twitter.com/en/docs/direct-messages/sending-and-receiving/api-reference/list-events

Comment (bakayim) [reply tweepy/api.py]: Removed, thanks

**Comment by khurshid-alam** [comment]
I got 2 direct messages in past 30 days. How did I print exact json data of each direct message I am getting with this? Or is it still not possible with library ?

**Comment by bakayim** [comment]
> I got 2 direct messages in past 30 days. How did I print exact json data of each direct message I am getting with this? Or is it still not possible with library ?

I shared possible usage script above in Python. Variable, direct_messages, is a list of all messages. You can iterate on it and get the exact json data.

**Comment by khurshid-alam** [comment]
Thanks. I will take a look.

**Comment by bakayim** [comment]
> hi @bakayim
> can u give an example how to parse the output of direct_messages() ?

direct_messages variable is a dictionary so you can think of it as JSON

**Comment by b3yond** [comment]
What is necessary to merge this? We need this in our project. Can we help you somehow?

**Comment by bakayim** [comment]
> What is necessary to merge this? We need this in our project. Can we help you somehow?

Hello! If I am not wrong, tests are not passed. May be I am the reason may be tests are outdated. If you want to help it would be great to look at test why not passed. Since I am currently working full time and continuing to education I have no time to check this out. When it is merged, will be working fine. I am currently using this changes in production environment without any problem.

Thanks

**Comment by dcsena** [comment]
any update on this? Would be great to get this merged in.

**Comment by FancyJesse** [comment]
Any way to receive the DMs through on_data() function of a stream?

File: tweepy/models.py [hunk 2]
```diff
@@ -207,7 +207,7 @@ class DirectMessage(Model):
     @classmethod
     def parse(cls, api, json):
         dm = cls(api)
-        for k, v in json.items():
+        for k, v in pyjson.loads(dm._api.last_response.text).items():
```
Comment (Harmon758) [inline tweepy/models.py]: I'm not sure what the reasoning is in getting the JSON response in such a roundabout way.

**Comment by Harmon758** [comment]
Superceded by #1228.

**Comment by Harmon758** [comment]
@FancyJesse That is unrelated to this PR.
User Streams have been deprecated and retired in favor of the new Account Activity API.
For Account Activity API support, see #1075.

Also see:
https://developer.twitter.com/en/docs/accounts-and-users/subscribe-account-activity/migration/us-ss-migration-guide.html

https://blog.twitter.com/developer/en_us/topics/tools/2018/enabling-all-developers-to-build-on-the-account-activity-api.html
https://blog.twitter.com/developer/en_us/topics/tools/2017/announcing-more-functionality-to-improve-customer-engagements-on-twitter.html
https://blog.twitter.com/developer/en_us/topics/tools/2017/building-the-future-of-the-twitter-api-platform.html

https://twittercommunity.com/t/details-and-what-to-expect-from-the-api-deprecations-this-week-on-august-16-2018/110746
https://twittercommunity.com/t/reminder-site-streams-user-streams-and-legacy-dm-endpoints-will-be-sunset-august-16th/109482
https://twittercommunity.com/t/building-on-site-streams-user-streams-or-direct-messages-action-required-to-preserve-and-prevent-disruption-to-your-twitter-integration/105736
https://twittercommunity.com/t/date-for-full-deprecation-of-user-streams-site-streams-and-legacy-direct-message-api-endpoints/98734

**Pull request closed without merging**